### PR TITLE
FeedKit

### PR DESCRIPTION
**Pull request by mitchdowney**

### CODE REVIEW

File: Podverse/PVFeedParser.swift [hunk 1]
```diff
@@ -35,251 +34,448 @@ class PVFeedParser {
...
+        }
+        
+        if let link = feed.links?.first?.attributes?.href {
+            podcast.link = link
+        }
+        
+        if let imageUrlString = feed.logo, let imageUrl = URL(string:imageUrlString) {
+            podcast.imageUrl = imageUrl.absoluteString
+            do {
+                podcast.imageData = try Data(contentsOf: imageUrl)
```
Comment (kreonjr) [inline Podverse/PVFeedParser.swift:152]: Omg I had never noticed this. This is a synchronous method. Making the thread hang until its done. If any image takes even a little longer than it should, the whole thread stops until its done. Should we rethink of this perhaps?

File: Podverse/PVFeedParser.swift [hunk 2]
```diff
@@ -35,251 +34,448 @@ class PVFeedParser {
...
+        }
+
+        if let link = feed.link {
+            podcast.link = link
+        }
+
+        if let imageUrlString = feed.image?.url, let imageUrl = URL(string:imageUrlString) {
+            podcast.imageUrl = imageUrl.absoluteString
+            do {
+                podcast.imageData = try Data(contentsOf: imageUrl)
```
Comment (kreonjr) [inline Podverse/PVFeedParser.swift:268]: Same here. Synchronous method that probably hangs the thread

File: Podverse/PVFeedParser.swift [hunk 3]
```diff
@@ -35,251 +34,448 @@ class PVFeedParser {
...
-        if onlyGetMostRecentEpisode == true {
-            latestEpisodePubDate = item.feedPubDate
-            parser.abortParsing()
-            return
+        if let iTunesImageUrlString = feed.iTunes?.iTunesImage?.attributes?.href, let itunesImageURL = URL(string:iTunesImageUrlString) {
+            podcast.itunesImageUrl = itunesImageURL.absoluteString
+            do {
+                podcast.itunesImage = try Data(contentsOf: itunesImageURL)
+                if podcast.imageData == nil {
+                    podcast.imageData = try Data(contentsOf: itunesImageURL)
```
Comment (kreonjr) [inline Podverse/PVFeedParser.swift:280]: Same as above

File: Podverse/PVFeedParser.swift [hunk 4]
```diff
@@ -35,251 +34,448 @@ class PVFeedParser {
...
+        }
         
-        guard let feedUrl = self.feedUrl, let podcast = podcast else {
-            self.parsingPodcasts.podcastFinishedParsing()
-            DispatchQueue.main.async {
-                NotificationCenter.default.post(name: .feedParsingComplete, object: nil, userInfo: nil)
+        if let imageUrlString = feed.icon, let imageUrl = URL(string:imageUrlString) {
+            podcast.imageUrl = imageUrl.absoluteString
+            do {
+                podcast.imageData = try Data(contentsOf: imageUrl)
```
Comment (kreonjr) [inline Podverse/PVFeedParser.swift:400]: Same here

File: Podverse/PVFeedParser.swift (hunk 1 above)
Comment (mitchdowney) [reply Podverse/PVFeedParser.swift:152]: Ahhh good catch. Hmm yeah we should do something about this but I dunno what :#

File: Podverse/PVFeedParser.swift (hunk 4 above)
Comment (mitchdowney) [reply Podverse/PVFeedParser.swift:400]: If this is a tougher problem to solve, I propose we fix it on its own branch so we can still merge the feedKit in. feedKit seems to be working way faster than the last parser :D

**Pull request merged**